Makefile

```diff
@@ -74,7 +67,7 @@ image: $(DEFAULTORG)/bpftool
 $(DEFAULTORG)/bpftool: register
 	# Make sure we re-pull the base image to pick up security fixes.
 	# Limit the build to use only one CPU, This helps to work around qemu bugs such as https://bugs.launchpad.net/qemu/+bug/1098729
-	docker build $(DOCKER_BUILD_ARGS) --build-arg KERNEL_REF=$(KERNELREF) --build-arg KERNEL_REPO=$(KERNELREPO) --cpuset-cpus 0 --pull -t $(ARCHIMAGE) -f $(DOCKERFILE) .
+	docker build $(DOCKER_BUILD_ARGS) --platform=linux/$(ARCH) --build-arg SOURCE_REF=$(SOURCEREF) --build-arg SOURCE_REPO=$(SOURCEREPO) --cpuset-cpus 0 --pull -t $(ARCHIMAGE) -f Dockerfile .
```

Comment:
Better to use the `docker buildx build` buildkit utils. Also, we can remove the `--cpuset-cpus 0` command line flag. I am not sure why we want to pin to cpu 0.

Reply:
The reason for pinning is mentioned in the line above it. Not sure, if it's valid. If you know it's not relevant anymore, I can remove it.

Reply:
@hjiawei about using `buildx`, I tried it, and the build failed. For whatever reason, the compiled binary for other arch, were not compiled statically. If there is no important reason to switch, I prefer to not spend more time on it.

Reply:
I removed `--cpuset-cpus 0`, and also tried `buildx` one more time. It breaks building other arches. It does not worth spending more time on it.